model/variableSheet.cc

```diff
@@ -30,39 +31,44 @@ using ecolab::cairo::CairoSave;
 namespace minsky
 {
 	
-namespace
-{    
-  struct CroppedPango: public Pango
-  {
-    cairo_t* cairo;
-    double w, x=0, y=0;
-    CroppedPango(cairo_t* cairo, double width): Pango(cairo), cairo(cairo), w(width) {}
-    void setxy(double xx, double yy) {x=xx; y=yy;}
-    void show() {
-      CairoSave cs(cairo);
-      cairo_rectangle(cairo,x,y,w,height());
-      cairo_clip(cairo);
-      cairo_move_to(cairo,x,y);
-      Pango::show();
-    }
-  };
-}    
+  namespace
+  {    
+    struct CroppedPango: public Pango
+    {
+      cairo_t* cairo;
+      double w, x=0, y=0;
+      CroppedPango(cairo_t* cairo, double width): Pango(cairo), cairo(cairo), w(width) {}
+      void setxy(double xx, double yy) {x=xx; y=yy;}
+      void show() {
+        CairoSave cs(cairo);
+        cairo_rectangle(cairo,x,y,w,height());
+        cairo_clip(cairo);
+        cairo_move_to(cairo,x,y);
+        Pango::show();
+      }
+    };
+  }
 
-void VariableSheet::redraw(int, int, int width, int height)
-{
-  cairo_t* cairo=surface->cairo();
-  CroppedPango pango(cairo, colWidth);
-  rowHeight=15;
-  pango.setFontSize(5.0*rowHeight);
-  
+  void VariableSheet::redraw(int, int, int width, int height)
   {
-    Pango pango(cairo);
-    pango.setText("Watch this space");
-    cairo_move_to(cairo,800,400);
-    pango.show();
-    
-  }
-}
+    if (surface.get()) {
+        cairo_t* cairo=surface->cairo();  
+        CroppedPango pango(cairo, colWidth);
+        rowHeight=15;
+        pango.setFontSize(5.0*rowHeight);
+	    
+        if (!minsky().canvas.model->empty()) {	  
+          populateItemVector();			        
+          cairo_translate(cairo,offsx,offsy);           
+          draw(cairo);         
+          ecolab::cairo::Surface surf
+            (cairo_recording_surface_create(CAIRO_CONTENT_COLOR_ALPHA,NULL));
+          draw(surf.cairo());
+          m_width=surf.width();
+          m_height=surf.height();              
+        }     
+      }
+    }
```

Comment:
This method is a candidate for being dried out by moving it to the common base class

Reply:
Done.